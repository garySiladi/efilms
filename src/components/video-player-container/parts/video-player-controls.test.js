// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import VideoPlayerInterface from './video-player-controls';

test('Video is playing + selected postition is 0', () => {
  const tree : string = renderer.create(
    <VideoPlayerInterface
      episodeTitle={'Whatever'}
      isVideoPlaying
      selectedPosition={0}
      progress={0}
      currentTime={'10:10'}
      endOfVideo={'20:20'}
      isVisible
      isBackButtonSelected
    />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video is playing + selected postition is 1', () => {
  const tree : string = renderer.create(
    <VideoPlayerInterface
      episodeTitle={'Whatever'}
      isVideoPlaying
      selectedPosition={1}
      progress={0}
      currentTime={'10:10'}
      endOfVideo={'20:20'}
      isVisible
      isBackButtonSelected
    />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video is playing + selected postition is 2', () => {
  const tree : string = renderer.create(
    <VideoPlayerInterface
      episodeTitle={'Whatever'}
      isVideoPlaying
      selectedPosition={2}
      progress={0}
      currentTime={'10:10'}
      endOfVideo={'20:20'}
      isVisible
      isBackButtonSelected
    />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video is not playing ', () => {
  const tree : string = renderer.create(
    <VideoPlayerInterface
      episodeTitle={'Whatever'}
      isVideoPlaying={false}
      selectedPosition={0}
      progress={0}
      currentTime={'10:10'}
      endOfVideo={'20:20'}
      isVisible
      isBackButtonSelected
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
