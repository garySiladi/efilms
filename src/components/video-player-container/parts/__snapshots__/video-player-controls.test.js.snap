// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video is not playing  1`] = `
<div
  className="player-interface"
>
  <div
    className="player-interface__top"
  >
    <button
      className="player-interface__back-button selected"
    >
      <span
        aria-hidden={true}
        className="fa fa-chevron-left player-interface__navigation-icons player-interface__navigation-icons--back"
      />
    </button>
    <span
      className="player-interface__episode-title"
    >
      Whatever
    </span>
  </div>
  <div
    className="player-interface__bottom"
  >
    <div
      className="player-interface__progress-time"
    >
      <div
        className="player-interface__progress-time--current"
      >
        10:10
      </div>
      <div
        className="player-interface__progress-time--final"
      >
        20:20
      </div>
    </div>
    <div
      className="player-interface__progress-bar-container"
    >
      <div
        className="player-interface__progress-bar"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
    <div
      className="player-interface__navigation-wrapper"
    >
      <button
        className="player-interface__navigation-control selected"
      >
        <span
          aria-hidden={true}
          className="fa fa-step-backward player-interface__navigation-icons player-interface__navigation-icons--step-backward"
          id="1"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-backward player-interface__navigation-icons player-interface__navigation-icons--rewind"
          id="2"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-play player-interface__navigation-icons player-interface__navigation-icons--play"
          id="3"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-forward player-interface__navigation-icons player-interface__navigation-icons--forward"
          id="4"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-step-forward player-interface__navigation-icons player-interface__navigation-icons--step-forward"
          id="5"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Video is playing + selected postition is 0 1`] = `
<div
  className="player-interface"
>
  <div
    className="player-interface__top"
  >
    <button
      className="player-interface__back-button selected"
    >
      <span
        aria-hidden={true}
        className="fa fa-chevron-left player-interface__navigation-icons player-interface__navigation-icons--back"
      />
    </button>
    <span
      className="player-interface__episode-title"
    >
      Whatever
    </span>
  </div>
  <div
    className="player-interface__bottom"
  >
    <div
      className="player-interface__progress-time"
    >
      <div
        className="player-interface__progress-time--current"
      >
        10:10
      </div>
      <div
        className="player-interface__progress-time--final"
      >
        20:20
      </div>
    </div>
    <div
      className="player-interface__progress-bar-container"
    >
      <div
        className="player-interface__progress-bar"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
    <div
      className="player-interface__navigation-wrapper"
    >
      <button
        className="player-interface__navigation-control selected"
      >
        <span
          aria-hidden={true}
          className="fa fa-step-backward player-interface__navigation-icons player-interface__navigation-icons--step-backward"
          id="1"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-backward player-interface__navigation-icons player-interface__navigation-icons--rewind"
          id="2"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-pause player-interface__navigation-icons player-interface__navigation-icons--pause"
          id="3"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-forward player-interface__navigation-icons player-interface__navigation-icons--forward"
          id="4"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-step-forward player-interface__navigation-icons player-interface__navigation-icons--step-forward"
          id="5"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Video is playing + selected postition is 1 1`] = `
<div
  className="player-interface"
>
  <div
    className="player-interface__top"
  >
    <button
      className="player-interface__back-button selected"
    >
      <span
        aria-hidden={true}
        className="fa fa-chevron-left player-interface__navigation-icons player-interface__navigation-icons--back"
      />
    </button>
    <span
      className="player-interface__episode-title"
    >
      Whatever
    </span>
  </div>
  <div
    className="player-interface__bottom"
  >
    <div
      className="player-interface__progress-time"
    >
      <div
        className="player-interface__progress-time--current"
      >
        10:10
      </div>
      <div
        className="player-interface__progress-time--final"
      >
        20:20
      </div>
    </div>
    <div
      className="player-interface__progress-bar-container"
    >
      <div
        className="player-interface__progress-bar"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
    <div
      className="player-interface__navigation-wrapper"
    >
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-step-backward player-interface__navigation-icons player-interface__navigation-icons--step-backward"
          id="1"
        />
      </button>
      <button
        className="player-interface__navigation-control selected"
      >
        <span
          aria-hidden={true}
          className="fa fa-backward player-interface__navigation-icons player-interface__navigation-icons--rewind"
          id="2"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-pause player-interface__navigation-icons player-interface__navigation-icons--pause"
          id="3"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-forward player-interface__navigation-icons player-interface__navigation-icons--forward"
          id="4"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-step-forward player-interface__navigation-icons player-interface__navigation-icons--step-forward"
          id="5"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Video is playing + selected postition is 2 1`] = `
<div
  className="player-interface"
>
  <div
    className="player-interface__top"
  >
    <button
      className="player-interface__back-button selected"
    >
      <span
        aria-hidden={true}
        className="fa fa-chevron-left player-interface__navigation-icons player-interface__navigation-icons--back"
      />
    </button>
    <span
      className="player-interface__episode-title"
    >
      Whatever
    </span>
  </div>
  <div
    className="player-interface__bottom"
  >
    <div
      className="player-interface__progress-time"
    >
      <div
        className="player-interface__progress-time--current"
      >
        10:10
      </div>
      <div
        className="player-interface__progress-time--final"
      >
        20:20
      </div>
    </div>
    <div
      className="player-interface__progress-bar-container"
    >
      <div
        className="player-interface__progress-bar"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
    <div
      className="player-interface__navigation-wrapper"
    >
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-step-backward player-interface__navigation-icons player-interface__navigation-icons--step-backward"
          id="1"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-backward player-interface__navigation-icons player-interface__navigation-icons--rewind"
          id="2"
        />
      </button>
      <button
        className="player-interface__navigation-control selected"
      >
        <span
          aria-hidden={true}
          className="fa fa-pause player-interface__navigation-icons player-interface__navigation-icons--pause"
          id="3"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-forward player-interface__navigation-icons player-interface__navigation-icons--forward"
          id="4"
        />
      </button>
      <button
        className="player-interface__navigation-control"
      >
        <span
          aria-hidden={true}
          className="fa fa-step-forward player-interface__navigation-icons player-interface__navigation-icons--step-forward"
          id="5"
        />
      </button>
    </div>
  </div>
</div>
`;
